nifi-commons/nifi-security-utils/src/test/java/org/apache/nifi/security/util/crypto/AESKeyedCipherProviderTest.java

```diff
@@ -14,311 +14,298 @@
+import org.junit.jupiter.api.Test;
 
-import javax.crypto.Cipher
-import javax.crypto.SecretKey
-import javax.crypto.spec.SecretKeySpec
-import java.security.SecureRandom
-import java.security.Security
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.SecretKeySpec;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.Security;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
-import static org.junit.jupiter.api.Assertions.assertEquals
-import static org.junit.jupiter.api.Assertions.assertFalse
-import static org.junit.jupiter.api.Assertions.assertThrows
-import static org.junit.jupiter.api.Assertions.assertTrue
-import static org.junit.jupiter.api.Assumptions.assumeTrue
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
-class AESKeyedCipherProviderGroovyTest {
-    private static final Logger logger = LoggerFactory.getLogger(AESKeyedCipherProviderGroovyTest.class)
+public class AESKeyedCipherProviderTest {
+    private static final String KEY_HEX = "0123456789ABCDEFFEDCBA9876543210";
 
-    private static final String KEY_HEX = "0123456789ABCDEFFEDCBA9876543210"
+    private static final String PLAINTEXT = "ExactBlockSizeRequiredForProcess";
 
-    private static final String PLAINTEXT = "ExactBlockSizeRequiredForProcess"
+    private static final List<EncryptionMethod> keyedEncryptionMethods = Arrays.stream(EncryptionMethod.values())
+            .filter(EncryptionMethod::isKeyedCipher)
+            .collect(Collectors.toList());
 
-    private static final List<EncryptionMethod> keyedEncryptionMethods = EncryptionMethod.values().findAll { it.keyedCipher }
+    private static SecretKey key;
 
-    private static final SecretKey key = new SecretKeySpec(Hex.decodeHex(KEY_HEX as char[]), "AES")
 
     @BeforeAll
     static void setUpOnce() throws Exception {
-        Security.addProvider(new BouncyCastleProvider())
+        Security.addProvider(new BouncyCastleProvider());
 
-        logger.metaClass.methodMissing = { String name, args ->
-            logger.info("[${name?.toUpperCase()}] ${(args as List).join(" ")}")
+        try {
+            key = new SecretKeySpec(Hex.decodeHex(KEY_HEX.toCharArray()), "AES");
+        } catch (final DecoderException e) {
+            throw new RuntimeException(e);
         }
     }
 
     private static boolean isUnlimitedStrengthCryptoAvailable() {
-        Cipher.getMaxAllowedKeyLength("AES") > 128
+        try {
+            return Cipher.getMaxAllowedKeyLength("AES") > 128;
+        } catch (final NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
+        }
     }
```

Comment:
This check is no longer needed, so the method can be removed and anything that depends on it can be adjusted to always test unlimited strength values.

Reply:
Will remove it